src/lib.rs

```diff
@@ -277,7 +275,16 @@ impl Metadata {
     /// ```
     pub fn new_from_path<S: AsRef<ffi::OsStr>>(path: S) -> Result<Metadata> {
         let mut err: *mut gexiv2::GError = ptr::null_mut();
-        let c_str_path = ffi::CString::new(path.as_ref().as_bytes()).unwrap();
+
+        // let c_str_path = ffi::CString::new(path.as_ref().as_bytes()).unwrap();
+        let c_str_path = ffi::CString::new(
+            path
+            .as_ref()
+            .to_str()
```

Comment:
This will do a check for valid UTF-8, as paths aren't necessarily encoded that way, but Rust strings are. This will therefore fail on paths that are valid in the host system, but which aren't UTF-8.

While I wish that everyone were using UTF-8 everywhere, I sadly suspect that's not the case. However if we can't find a better way it's a tradeoff I'm OK with making -- we'd be consciously choosing to only support UTF-8, though.

That said, I believe Windows paths are actually UTF-16... so I'm not sure if they'll work at all either?